### ISSUE TITLE
HtmlWidgetConfig.onTapUrl is called, but HtmlWidget.onTapUrl isn't.

### ISSUE THREAD

**Issue by red42** [daohoangson/flutter_widget_from_html#184]
I'm trying to handle custom URLs on my HTML strings, but I can't get the widget's onTapUrl to be called.

```dart
HtmlWidgetConfig defaultHtmlConfig = HtmlWidgetConfig(
    bodyPadding: const EdgeInsets.all(0),
    textStyle: TextStyle(fontSize: 13),
    onTapUrl: (url) { print('Tapped $url from config'); },    
);

HtmlWidget(
  'Click <a href="myscheme://link">here</a>',
  config: Styles.defaultHtmlConfig,
  onTapUrl: (url) { print('Tapped $url from widget'); },
);
```

The first onTapUrl gets called as expected, but the second one isn't. If I don't set onTapUrl on config, I get `[flutter_widget_from_html] Tapped url myscheme://link` on the console.

Looking at the code, I found the following bit:
```dart
  GestureTapCallback buildGestureTapCallbackForUrl(String url) => url != null
      ? () => _config.onTapUrl != null
          ? _config.onTapUrl(url)
          : print("[flutter_widget_from_html] Tapped url $url")
      : null;
```

Which is consistent with the behaviour I'm seeing, but I couldn't find the code where the widget's `onTapUrl` would be called.

**Comment by red42** [comment]
Upon further inspection, I see the widget's onTapUrl is used on a default HtmlWidgetConfig only if a config value isn't passed.

```dart
  HtmlWidget(
    this.html, {
    this.enableCaching = true,
    this.factoryBuilder,
    Key key,
    HtmlWidgetConfig config,
    Uri baseUrl,
    EdgeInsets bodyPadding = const EdgeInsets.all(10),
    NodeMetadataCollector builderCallback,
    Color hyperlinkColor = const Color.fromRGBO(0, 0, 255, 1),
    OnTapUrl onTapUrl,
    EdgeInsets tableCellPadding = const EdgeInsets.all(5),
    TextStyle textStyle = const TextStyle(),
  })  : assert(html != null),
        _config = config ??
            HtmlWidgetConfig(
              baseUrl: baseUrl,
              bodyPadding: bodyPadding,
              builderCallback: builderCallback,
              hyperlinkColor: hyperlinkColor,
              onTapUrl: onTapUrl,
              tableCellPadding: tableCellPadding,
              textStyle: textStyle,
            ),
        super(key: key);
```

**Comment by red42** [comment]
The above gives me what I need. But may I suggest either merging the config and widget values on the constructor, or having a .copyWith() constructor on the config?

**Comment by daohoangson** [comment]
Yeah, it was confusing with 2 different ways to configure so I have dropped the `config` param all together. This will be included in v0.4.0 (as it is a breaking change). Thank you for the suggestion.

**Comment by red42** [comment]
Ah, I see. That's great to hear, I'll prepare my code for that change!
